### PR TITLE
add generateArgVerification 

### PR DESCRIPTION
**Pull request by Elli610**
related to #207

### CODE REVIEW

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ProtoFile } from './protobuf';
+
+function generateArgVerification(protoFile: ProtoFile): string {
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:3]: What is you function doing ?
If it generates code to check if an unsigned argument of a protobuf message are negative, you should rename it to something easier to understand.

Furthermore, please add comments.

My suggestion based on what I undestand:
```suggestion
/**
 * Generates code to check if unsigned arguments of a protobuf message are negative.
 * @param {ProtoFile} protoFile - The protofile object.
 * @return {string} The string containing code for unsigned argument checks.
 */
function generateUnsignedArgCheckCode(protoFile: ProtoFile): string {
```

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  let content = '';
+  for (let arg of protoFile.argFields) {
+    if (
+      arg.type === 'uint32' ||
+      arg.type === 'uint64' ||
+      arg.type === 'fixed32' ||
+      arg.type === 'fixed64'
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:10]: You can simplify/clarify your code by regrouping all protobu unsigned int in an array and check that this arg.type is not in this array.

Something like:
```ts
  const unsignedPBTypes = new Set(['uint32', 'uint64', 'fixed32', 'fixed64']);
  ...
    if (unsignedPBTypes.has(arg.type)) {
```

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    ) {
+      content += `\tif(${arg.name} < 0){\n`;
+      content += `\t\tthrow new Error("Invalid argument: ${arg.name} cannot be negative");\n`;
+      content += `\t}\n`;
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:14]: I personaly think that one template litteral is easier to read:
```suggestion
      content +=`
  if (${arg.name} < 0) {
    throw new Error("Invalid argument: ${arg.name} cannot be negative accordind to protobuf file.");
  }`;
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Generates code to check if unsigned arguments of a protobuf message are negative.
+ *
+ * @param protoFile - The protofile object.
+ * @returns The string containing code for unsigned argument checks.
+ */
+function generateUnsignedArgCheckCode(protoFile: ProtoFile): string {
+  const unsignedPBTypes = new Set(['uint32', 'uint64', 'fixed32', 'fixed64']);
+  let content = '';
+  for (let arg of protoFile.argFields) {
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:12]:
```suggestion
  for (const arg of protoFile.argFields) {
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/massa-proto-cli/src/ts-gen.ts (hunk 4 above)
Comment (Elli610) [reply packages/massa-proto-cli/src/ts-gen.ts:12]: fixed

**Pull request closed without merging**